scalanet/src/io/iohk/scalanet/peergroup/kademlia/KBuckets.scala

```diff
@@ -0,0 +1,109 @@
+package io.iohk.scalanet.peergroup.kademlia
+
+import java.util.concurrent.{ConcurrentHashMap, CopyOnWriteArraySet}
+
+import io.iohk.scalanet.peergroup.kademlia.KBuckets._
+import scodec.bits.BitVector
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+
+class KBuckets(val baseId: BitVector) {
+
+  private val buckets = new ConcurrentHashMap[Int, KBucket]().asScala
+
+  add(baseId)
```

Comment:
minor comment, in the paper, the node does not store its own id in its KBuckets.

Reply:
True. I couldn't see a reason why not and it made the code a bit cleaner to store it.

Reply:
See also other comment about this below.